src/core/theme/utils/defaultTheme.ts

```diff
@@ -1,12 +1,21 @@
 import { suomifiTheme, ThemeProp } from '../';
+import { asPropType } from '../../../utils/typescript';
 
+/**
+ * Do defaults for component props
+ * Add default theme if theme not defined,
+ * Remove as-prop and change it to asProp if defined
+ * @param props Component props
+ */
 export const withDefaultTheme = <
-  T extends { theme: ThemeProp; [k: string]: any }
+  T extends { theme: ThemeProp; asProp?: asPropType }
 >({
-  theme = suomifiTheme,
+  theme,
+  as,
   ...props
-}: Partial<T>): T =>
+}: Partial<T> & { as?: asPropType }): T =>
   ({
     ...props,
-    theme,
+    ...(!!as ? { asProp: as } : {}),
```

Comment:
Our HTML-components have `as`-prop defined as they use `style` from styled-components that can handle element/component change with it. The type definition for it is traversed with `extended` to our final UI-components.
https://www.styled-components.com/docs/api#as-polymorphic-prop

Our UI-components use this defaultTheme-function to set defaults for theme. I'm implementing here that at the same time that prop named `as` is taken away and give as `asProp` - so that every time our internal implementation uses `styled` it does not change the internal component it has. Then we need to internally make implementation for `asProp` where we like to use it - otherwise it does nothing even when defined (originally `as`).